GameData/RP-1/Contracts/Commercial Applications 2/MolniyaNetwork3.cfg

```diff
@@ -0,0 +1,367 @@
+
+	description = <b>Program: Advanced Commercial Applications<br>Type: <color=green>Required</color></b><br><br>Geostationary satellites do not provide good coverage in the higher latitudes.  Molniya orbits loiter over those latitudes for longer periods, to provide constant coverage with fewer satellites. <br>Historical example: Molniya 1-1 (1.6t, Molniya 8K78) launched starting in April of  1965.<br><br><b><color=yellow>NOTE: The satellite will be destroyed upon completion of the contract. This simulates transfer of the payload back to the customer.</color><br><br><b><color=yellow>NOTE: Your orbit should have high eccentricity, an inclination of around 63 degrees, with your periapsis on the same or opposite longitude as the target (@/targetLon) in the opposite North/South hemisphere.  If you are completing this with a single launch, you must separate the 3 vessels before entering the specified orbit.</color>
+
+	synopsis = Launch a 3 Satellite Molniya Communications Network
+
+	completedMessage = Congratulations! The satellites are in orbit and spaced to provide complete coverage.
+
+	sortKey = 906
+
+	cancellable = true
+	declinable = false
+	autoAccept = false
+	minExpiry = 0
+	maxExpiry = 0
+	maxCompletions = 1
+	maxSimultaneous = 1
+	deadline = 0
+	targetBody = HomeWorld()
+
+	// ************ REWARDS ************
+	prestige = Trivial       // 1.0x
+	advanceFunds = 0
+	rewardScience = 0
+	rewardFunds = 0
+	failureFunds = 0
+	rewardReputation = 160
+	failureReputation = 0 // was @rewardReputation
+
+	// ************ REQUIREMENTS ************
+
+	REQUIREMENT
+	{
+		name = ProgramActive
+		type = ProgramActive
+		program = CommercialApplications2
+	}
+	REQUIREMENT
+	{
+		name = CompleteContract
+		type = CompleteContract
+		contractType = FirstMolniyaSat-CA
+	}
+	REQUIREMENT
+	{
+		name = AcceptContract
+		type = AcceptContract
+		contractType = EarlyMolniyaSat
+		invertRequirement = true
+	}
+
+	BEHAVIOUR
+	{
+		name = WaypointGenerator
+		type = WaypointGenerator
+
+		WAYPOINT
+		{
+			name = Molniya Network Target
+			icon = thermometer
+			altitude = 0
+			latitude = @/targetLat
+        	longitude = @/targetLon
+		}
+	}
+
+	DATA
+	{
+		type = int
+		targetLon = Random(-180,180)
+		targetLat = [63,-63].Random()
+	}
+
+	PARAMETER
+	{
+		name = MolniyaCom1
+		type = VesselParameterGroup
+		define = Molniya Network Sat I
+
+		PARAMETER
+		{
+			name = IsNotVessel
+			type = IsNotVessel
+
+			vessel = Molniya Network Sat II
+		}
+
+		PARAMETER
+		{
+			name = IsNotVessel
+			type = IsNotVessel
+
+			vessel = Molniya Network Sat III
+		}
+
+		PARAMETER
+		{
+			name = NewVessel
+			type = NewVessel
+			title = Launch a new vessel
+			hideChildren = true
+		}
+		PARAMETER
+		{
+			name = Crewmembers
+			type = HasCrew
+			minCrew = 0
+			maxCrew = 0
+			title = Uncrewed
+			hideChildren = true
+		}
+		PARAMETER
+		{
+			name = HasComSatPayload
+			type = HasResource
+			resource = ComSatPayload
+			minQuantity = 124.9
+			maxQuantity = 249.0
```

Comment:
Is the intention of max to prevent double-dipping with another contract? Since you overrode the title then the existence of this upper limit isn't communicated to the player in any way.

Reply:
Ideally a Molniya network would be spaced out for constant coverage, but there's not a way to check for that.  Instead, I opted to at least make sure the player doesn't launch all 3 on one vessel and then decouple them into identical orbits.  This allows you to launch them all at once if you want to, but requires you to decouple before finalizing your orbit which will give at least some spacing.  
I'm not sure how strongly people feel about preventing the first case, but if we want to keep this check I should definitely clarify the contract.